core/src/commonMain/kotlin/com/xebia/functional/xef/llm/assistants/AssistantThread.kt

```diff
@@ -0,0 +1,242 @@
+package com.xebia.functional.xef.llm.assistants
+
+import com.xebia.functional.openai.apis.AssistantsApi
+import com.xebia.functional.openai.infrastructure.ApiClient
+import com.xebia.functional.openai.models.*
+import com.xebia.functional.openai.models.ext.assistant.RunStepDetailsMessageCreationObject
+import com.xebia.functional.openai.models.ext.assistant.RunStepDetailsToolCallsObject
+import com.xebia.functional.openai.models.ext.assistant.RunStepObjectStepDetails
+import com.xebia.functional.xef.llm.fromEnvironment
+import kotlin.jvm.JvmName
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.flow
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+
+class AssistantThread(
+  private val threadId: String,
+  private val api: AssistantsApi = fromEnvironment(::AssistantsApi)
+) {
+
+  suspend fun delete(): Boolean = api.deleteThread(threadId).body().deleted
+
+  suspend fun modify(request: ModifyThreadRequest): AssistantThread =
+    AssistantThread(api.modifyThread(threadId, request).body().id)
+
+  suspend fun createMessage(message: MessageWithFiles): MessageObject =
+    api
+      .createMessage(
+        threadId,
+        CreateMessageRequest(
+          role = CreateMessageRequest.Role.user,
+          content = message.content,
+          fileIds = message.fileIds
+        )
+      )
+      .body()
+
+  suspend fun createMessage(content: String): MessageObject =
+    api
+      .createMessage(
+        threadId,
+        CreateMessageRequest(role = CreateMessageRequest.Role.user, content = content)
+      )
+      .body()
+
+  suspend fun createMessage(request: CreateMessageRequest): MessageObject =
+    api.createMessage(threadId, request).body()
+
+  suspend fun getMessage(messageId: String): MessageObject =
+    api.getMessage(threadId, messageId).body()
+
+  suspend fun listMessages(): List<MessageObject> = api.listMessages(threadId).body().data
+
+  suspend fun createRun(request: CreateRunRequest): RunObject =
+    api.createRun(threadId, request).body()
+
+  suspend fun getRun(runId: String): RunObject =
+    api
+      .getRun(threadId, runId)
+      .also {
+        //  it.response.bodyAsText().let(::println)
```

Comment:
This has been left here after a crazy debugging? And the same for the next code commented in this file.